### PR TITLE
Text teaser component with themed background

### PR DESCRIPTION
**Pull request by padms**

### CODE REVIEW

File: sanityv3/schemas/objects/textTeaser.tsx [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+      validation: (Rule: Rule) => Rule.max(1).error('Only one action is permitted'),
+    },
+    {
+      name: 'titlePosition',
+      title: 'Title position',
+      description: 'Select which side of the teaser the title should be displayed at on larger screens.',
+      type: 'string',
+      fieldset: 'design',
+      components: {
+        input: function ({ onChange, value }: { onChange: any; value: string }) {
```
Comment (fernandolucchesi) [inline sanityv3/schemas/objects/textTeaser.tsx:108]: any on sanityv3 is bad... we should also refactor the RadioIconSelector...

`onChange: (event: PatchEvent) => void`

File: sanityv3/schemas/objects/textTeaser.tsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+  preview: {
+    select: {
+      title: 'title',
+      text: 'text',
+    },
+    prepare({ title, text }: { title: PortableTextBlock[]; text: PortableTextBlock[] }) {
+      const plainTitle = blocksToText(title)
+      return {
+        title: plainTitle || 'Missing teaser title',
+        subtitle: blocksToText(text) || 'Mising teaser text',
```
Comment (fernandolucchesi) [inline sanityv3/schemas/objects/textTeaser.tsx:138]: Would recommentd prefixing the string 'Text Teaser | ' before adding subtitle content.. helps editors visualizing what section it is before going into the component. Also nice to add a media if you can find an interesting icon

File: web/lib/queries/common/pageContentFields.ts [hunk 3]
```diff
@@ -38,6 +38,22 @@ const pageContentFields = /* groq */ `
       ${downloadableImageFields},
     },
   },
+
+  _type == "textTeaser" => {
```
Comment (fernandolucchesi) [inline web/lib/queries/common/pageContentFields.ts:42]: extract into its own file as we did for the latest added components - so we have a readable/scalable groq in the future.

File: web/lib/queries/common/pageContentFields.ts [hunk 4]
```diff
@@ -38,6 +38,22 @@ const pageContentFields = /* groq */ `
...
+    "type": _type,
+    "id": _key,
+    title,
+    "text": text[]{..., ${markDefs}},
+    "designOptions": {
+      "theme": coalesce(theme.value, 0),
```
Comment (fernandolucchesi) [inline web/lib/queries/common/pageContentFields.ts:48]: Again, not a fan of the 0, especially when reading it like this (imagine you have no context of the component, is reading the query, and sees that default value is 0, you get not value from it)

File: web/pageComponents/shared/textTeaser/TextTeaser.tsx [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+import { getUrlFromAction } from '../../../common/helpers/getUrlFromAction'
+
+import type { TextTeaserData, LinkData } from '../../../types/types'
+import { getLocaleFromName } from '../../../lib/localization'
+import { getColorForTheme } from './theme'
+import { CSSProperties } from 'react'
+
+const { Content } = EnvisTeaser
+
+type TitlePostion = 'left' | 'right' | undefined
```
Comment (fernandolucchesi) [inline web/pageComponents/shared/textTeaser/TextTeaser.tsx:14]: undefined should not be an option

Comment (fernandolucchesi) [reply web/pageComponents/shared/textTeaser/TextTeaser.tsx:14]: `"titlePosition": coalesce(titlePosition, 'left')`

File: web/pageComponents/shared/textTeaser/TextTeaser.tsx [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+  --max-content-width: 1440px;
+  display: grid;
+  grid-template-columns: 1fr;
+  grid-template-rows: min-content min-content;
+  grid-template-areas:
+    'title'
+    'content';
+  max-width: var(--max-content-width);
+  margin: 0 auto;
+  /* Hardcoded value while waiting for the w3c proposal for environment() */
```
Comment (fernandolucchesi) [inline web/pageComponents/shared/textTeaser/TextTeaser.tsx:57]: what is this comment about?

File: web/pageComponents/shared/textTeaser/TextTeaser.tsx [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+    </StyledLink>
+  )
+}
+
+const TextTeaser = ({ data, anchor }: TextTeaserProps) => {
+  const { title, text, action, designOptions } = data
+  const { theme, titlePosition } = designOptions
+  const { background, highlight } = getColorForTheme(theme)
+
+  if ([title, text, action].every((i) => !i)) {
```
Comment (fernandolucchesi) [inline web/pageComponents/shared/textTeaser/TextTeaser.tsx:98]: What is this for?

File: web/pageComponents/shared/textTeaser/TextTeaser.tsx [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+
+const { Content } = EnvisTeaser
+
+type TitlePostion = 'left' | 'right' | undefined
+
+type TextTeaserProps = {
+  data: TextTeaserData
+  anchor?: string
+}
+
```
Comment (fernandolucchesi) [inline web/pageComponents/shared/textTeaser/TextTeaser.tsx:20]: somewhere in your styles an extra padding is being added to the last component:

image

File: web/types/types.ts [hunk 9]
```diff
@@ -306,6 +306,18 @@ export type TeaserData = {
...
 
+export type TextTeaserData = {
+  type: string
+  id: string
+  title: PortableTextBlock[]
+  text: PortableTextBlock[]
+  action?: LinkData
+  designOptions: {
+    theme: number
+    titlePosition?: 'left' | 'right' | undefined
```
Comment (fernandolucchesi) [inline web/types/types.ts:317]: no need for undefined

**Review by fernandolucchesi — CHANGES_REQUESTED** [review]
Looking nice @padms. Left some suggestion and few comments in the code.

File: sanityv3/schemas/components/ThemeSelector/defaultColors.ts [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+    case 7:
+      return { background: '#49709C', highlight: '#FFF5B8' }
+    case 8:
+      return { background: '#C3E4CE', highlight: 'hsl(348, 100%, 54%)' }
+
+    case 0:
+    default:
+      return { background: 'hsl(0, 0%, 100%)', highlight: 'hsl(348, 100%, 54%)' }
+  }
+}
```
[the remark this answers is not in the document]
Comment (padms) [reply sanityv3/schemas/components/ThemeSelector/defaultColors.ts:36]: I tried to do something like this before, but couldn't think of naming when there are multiple highlight colors for a single component.. so made it simple

**Review by fernandolucchesi — APPROVED** [review]

**Review by fernandolucchesi — APPROVED** [review]

**Pull request merged**